src/main/java/io/confluent/connect/hdfs/DataWriter.java

```diff
@@ -0,0 +1,333 @@
+/**
+ * Copyright 2015 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ **/
+
+package io.confluent.connect.hdfs;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.Path;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.errors.ConnectException;
+import org.apache.kafka.connect.sink.SinkRecord;
+import org.apache.kafka.connect.sink.SinkTaskContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Queue;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+
+import io.confluent.connect.avro.AvroData;
+import io.confluent.connect.hdfs.filter.CommittedFileFilter;
+import io.confluent.connect.hdfs.filter.TopicCommittedFileFilter;
+import io.confluent.connect.hdfs.hive.HiveMetaStore;
+import io.confluent.connect.hdfs.hive.HiveUtil;
+import io.confluent.connect.hdfs.partitioner.Partitioner;
+import io.confluent.connect.hdfs.storage.Storage;
+import io.confluent.connect.hdfs.storage.StorageFactory;
+
+public class DataWriter {
+  private static final Logger log = LoggerFactory.getLogger(DataWriter.class);
+
+  private Map<TopicPartition, TopicPartitionWriter> topicPartitionWriters;
+  private String url;
+  private Storage storage;
+  private Configuration conf;
+  private String topicsDir;
+  private Format format;
+  private Set<TopicPartition> assignment;
+  private Partitioner partitioner;
+  private RecordWriterProvider writerProvider;
+  private SchemaFileReader schemaFileReader;
+  private Map<TopicPartition, Long> offsets;
+  private Set<TopicPartition> lastAssignment;
+  private HdfsSinkConnectorConfig connectorConfig;
+  private AvroData avroData;
+  private SinkTaskContext context;
+  private ExecutorService executorService;
+  private String hiveDatabase;
+  private HiveMetaStore hiveMetaStore;
+  private HiveUtil hive;
+  private Queue<Future> futures;
```

Comment:
Futures for what? This names could probably be clearer -- maybe something like `hiveUpdateFutures`?